test/parsing/fixed-paths.test.js

```diff
@@ -1,42 +1,103 @@
+     ]
 
-  t.plan(1)
-  t.deepEqual(out,
-    [ root, { type: "fixed", input: "foo" } ]
-  )
+  t.deepEqual(rhumb._parse('/foo'), parts
+    , 'returns a root part and fixed part when being parsed')
+
+  t.deepEqual(rhumb._parse('/foo/'), parts
+    , 'returns a root part and fixed part when being parsed')
 })
 
 test("Parsing should find multiple fixed parts", function(t) {
-  var out = rhumb._parse("/foo/bar/bing")
-
-  t.plan(1)
-  t.deepEqual(out,
-    [ root
-    , { type: "fixed", input: "foo"  }
-    , { type: "fixed", input: "bar"  }
-    , { type: "fixed", input: "bing" }
-    ]
-  )
+  t.plan(2)
+
+  var parts = [
+        utils.rootPart()
+      , utils.fixedPart('foo')
+      , utils.fixedPart('bar')
+      , utils.fixedPart('bing')
+     ]
+
+  t.deepEqual(rhumb._parse('/foo/bar/bing'), parts
+    , 'returns a root part and three fixed parts when being parsed')
+
+  t.deepEqual(rhumb._parse('/foo/bar/bing/'), parts
+    , 'returns a root part and three fixed parts when being parsed')
+})
+
+test('Parsing should handle a path with no leading slash', function(t) {
+  t.plan(4)
+
+  var singleParts = [
+        utils.rootPart()
+      , utils.fixedPart('foo')
+      ]
+    , multipleParts = [
+        utils.rootPart()
+      , utils.fixedPart('foo')
+      , utils.fixedPart('bar')
+      , utils.fixedPart('bing')
+      ]
+
+  t.deepEqual(rhumb._parse('/foo'), singleParts
+    , 'returns a root part and fixed part when being parsed')
+
+  t.deepEqual(rhumb._parse('/foo/bar/bing'), multipleParts
+    , 'returns a root part and three fixed parts when being parsed')
+
+  t.deepEqual(rhumb._parse('/foo/'), singleParts
+    , 'returns a root part and fixed part when being parsed')
+
+  t.deepEqual(rhumb._parse('/foo/bar/bing/'), multipleParts
+    , 'returns a root part and three fixed parts when being parsed')
 })
 
-test("Parsing should find single fixed part for /", function(t) {
-  var out = rhumb._parse("/")
+test('Parsing should handle a path with empty parts', function(t) {
+  t.plan(4)
 
-  t.plan(1)
-  t.deepEqual(out,
-    [ root ]
-  )
-})
\ No newline at end of file
+  var emptyStartParts = [
+        utils.rootPart()
+      , utils.fixedPart('bar')
+      , utils.fixedPart('bing')
+      ]
+    , emptyMiddleParts = [
+        utils.rootPart()
+      , utils.fixedPart('foo')
+      , utils.fixedPart('bing')
+      ]
+    , emptyEndParts = [
+        utils.rootPart()
+      , utils.fixedPart('foo')
+      , utils.fixedPart('bar')
+      ]
+
+  t.deepEqual(rhumb._parse('//'), [utils.rootPart()]
+    , 'returns a root part when being parsed')
+
+  t.deepEqual(rhumb._parse('//bar/bing'), emptyStartParts
+    , 'returns a root part and two fixed parts when being parsed')
+
+  t.deepEqual(rhumb._parse('/foo//bing'), emptyMiddleParts
+    , 'returns a root part and two fixed parts when being parsed')
+
+  t.deepEqual(rhumb._parse('/foo/bar//'), emptyEndParts
+    , 'returns a root part and two fixed parts when being parsed')
```

Comment:
I probably need to change these tests.  I was expecting them to change as I update the interpolate code to pass the tests.